index.js

```diff
@@ -10,16 +10,27 @@ const fs = require('fs');
 
 const { createValuesObject, parseMetadataComments } = require('./lib/parser');
 const { checkKeys } = require('./lib/checker');
-const { combineMetadataAndValues, buildSectionsArrays, buildParamsToRenderList } = require('./lib/builder');
+const { combineMetadataAndValues, buildSectionsObjects, buildParamsToRenderList } = require('./lib/builder');
 const { insertReadmeTable, renderOpenAPISchema } = require('./lib/render');
 
-function getParameters(options) {
+/*
+* Parses the given values file for all metadata comments and returns an object containing
+* Returns an object with following structure:
+* {
+*   'parsedValues': [],              // contains all parameter related data
+*   'parsedSectionDescriptions': [], // contains key-value pairs of section name (key) and an array of lines for a description (value)
+* }
+*/
+function getParsingResults(options) {
   const valuesFilePath = options.values;
   const configPath = options.config ? options.config : `${__dirname}/config.json`;
   const config = require(configPath);
 
+  return parseMetadataComments(valuesFilePath, config);
+}
+
+function getParameters(valuesFilePath, valuesMetadata) {
   const valuesObject = createValuesObject(valuesFilePath);
-  const valuesMetadata = parseMetadataComments(valuesFilePath, config);
```

Comment:
This line was below the `valuesObject` intentionally so anyone reading the code can see the difference between both variables during the rest of the code, I would leave it here for consistent structure